utils/rand/rand.go

```diff
@@ -0,0 +1,169 @@
+package rand
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"errors"
+	"fmt"
+)
+
+// This package is a wrppaer around true RNG crypto/rand.
+// It implements useful tools using the true RNG and that
+// are not exported by the crypto/rand package.
+// This package does not implement any determinstic RNG (Pseudo RNG)
+// unlike the package flow-go/crypto/random.
+
+var randFailure = errors.New("crypto/rand failed")
+
+// returns a random uint64
+func Uint64() (uint64, error) {
```

Comment:
maybe simplify to `return Uint64n(math.MaxUint64)`

Reply:
The purpose of this package as mentioned in the description is to provide an alternative for `math/rand`. `math/rand` is a weak randomness that we should only use in test code (that's what my long PR actually cleans up).

Reply:
I misread your initial comment, my reply doesn't make sense. now I see what you mean 

> maybe simplify to  Uint64n(math.MaxUint64)
This doesn't work because it excludes the value `math.MaxUint64` itself from being an output, and that makes the distribution not uniform.

Reply:
ah, my bad